tests/test_meshtags.py

```diff
@@ -0,0 +1,284 @@
+            org_map = generate_reference_map(mesh, ft, mesh.comm, root)
+            org_maps.append(org_map)
+        else:
+            if MPI.COMM_WORLD.rank == root:
+                adios4dolfinx.write_meshtags(filename, mesh, ft, engine="BP4")
+                # Create map from mesh tag value to its corresponding index and midpoint
+                org_map = generate_reference_map(mesh, ft, MPI.COMM_SELF, root)
+                org_maps.append(org_map)
+        del ft
+    del mesh
+
+    MPI.COMM_WORLD.Barrier()
+    # Read mesh on testing communicator
+    new_mesh = adios4dolfinx.read_mesh(
+        read_comm, filename, engine="BP4", ghost_mode=read_mode
+    )
+    for dim in range(new_mesh.topology.dim + 1):
+        # Read meshtags on all processes if testing communicator has multiple ranks
+        # else read on root 0
+        if read_comm.size != 1:
+            new_ft = adios4dolfinx.read_meshtags(
+                filename, new_mesh, meshtag_name=f"entity_{dim}", engine="BP4"
+            )
+            # Generate meshtags map from mesh tag value to its corresponding index and midpoint
+            # and gather on root process
+            read_map = generate_reference_map(new_mesh, new_ft, new_mesh.comm, root)
+        else:
+            if MPI.COMM_WORLD.rank == root:
+                new_ft = adios4dolfinx.read_meshtags(
+                    filename, new_mesh, meshtag_name=f"entity_{dim}", engine="BP4"
+                )
+                read_map = generate_reference_map(new_mesh, new_ft, read_comm, root)
+
+        # On root process, check that midpoints are the same for each value in the meshtag
+        if MPI.COMM_WORLD.rank == root:
+            org_map = org_maps[dim]
+            assert len(org_map) == len(read_map)
+            for value, (_, midpoint) in org_map.items():
+                _, read_midpoint = read_map[value]
+                np.testing.assert_allclose(read_midpoint, midpoint)
+
+
+@pytest.mark.parametrize("read_mode", read_modes)
+@pytest.mark.parametrize("read_comm", [MPI.COMM_SELF, MPI.COMM_WORLD])
+def test_checkpointing_meshtags_2D(mesh_2D, read_comm, read_mode):
+    mesh = mesh_2D
+    hash = f"{mesh.comm.size}_{mesh.topology.cell_name()}_{mesh.geometry.x.dtype}"
+    filename = f"meshtags_1D_{hash}.bp"
+    if mesh.comm.size != 1:
+        adios4dolfinx.write_mesh(mesh, filename, engine="BP4")
+    else:
+        if MPI.COMM_WORLD.rank == root:
+            adios4dolfinx.write_mesh(mesh, filename, engine="BP4")
+
+    org_maps = []
+    for dim in range(mesh.topology.dim + 1):
+
+        mesh.topology.create_connectivity(dim, mesh.topology.dim)
+        e_map = mesh.topology.index_map(dim)
+        num_entities_local = e_map.size_local
+        entities = np.arange(num_entities_local, dtype=np.int32)
+        ft = dolfinx.mesh.meshtags(mesh, dim, entities, e_map.local_range[0] + entities)
+        ft.name = f"entity_{dim}"
```

Comment:
There is a lot of repetition between the tests. Not sure if it would be possible to refactor some of this into a helper function so that it becomes easier to maintain?

Reply:
Not sure its with the time. 
:P if you can figure out a way to Get the fixture meshes through parameterize it would be the best fix